### PR TITLE
Add regression test for net2cog

### PR DESCRIPTION
**Pull request by frankinspace**
## Description

Add regression test for net2cog

### CODE REVIEW

File: test/net2cog/net2cog_Regression.ipynb [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+    "        **non_prod_granule_data,\n",
+    "        'env': Environment.LOCAL,\n",
+    "    },\n",
+    "}\n",
+    "\n",
+    "environment_information = collection_data.get(harmony_host_url)\n",
+    "\n",
+    "if environment_information is not None:\n",
+    "    harmony_client = Client(\n",
+    "        env=environment_information['env'], auth=('greguska', 'r3gul@t0R!')\n",
```
[the remark this answers is not in the document]
Comment (flamingbear) [reply test/net2cog/net2cog_Regression.ipynb:138]: I took these out and I could use my creds, these may be 🔥'd

File: test/net2cog/utility.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
+"""Simple utility functions used in the regridder test notebook."""
```
Comment (flamingbear) [inline test/net2cog/utility.py:1]:
```suggestion
"""Simple utility functions used in the net2cog test notebook."""
```
I'm introducing shared utility directory in #92 but for now...

File: test/net2cog/net2cog_Regression.ipynb [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+    "            \"net2cog harmony adapter currently only supports processing one variable at a time\"\n",
+    "            in str(error)\n",
+    "        )\n",
+    "        raised_expected_error = True\n",
+    "        print(error)\n",
+    "\n",
+    "    assert (\n",
+    "        raised_expected_error\n",
+    "    ), 'Expected request to raise an exception but it did not.'"
+   ]
```
Comment (flamingbear) [inline test/net2cog/net2cog_Regression.ipynb:256]: would you mind adding a line at the bottom of the cell just for the humans?

```
    utility.print_success('All variables raised expected error')
```

**Review by flamingbear — CHANGES_REQUESTED** [review]
Looks great! I was able to run the tests interactively against Harmony-In-A-Box.

I do have a few small nits and suggestions for you before merging.

File: test/net2cog/environment.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+name: papermill-net2cog
+channels:
+  - conda-forge
+  - defaults
```
[the remark this answers is not in the document]
Comment (flamingbear) [reply test/net2cog/environment.yaml:4]: also I verified I can switch `- defaults` to `- nodefaults` and everything still worked correctly.

[the remark this answers is not in the document]
Comment (frankinspace) [reply test/net2cog/environment.yaml:4]: Thanks, updated

File: test/net2cog/net2cog_Regression.ipynb (hunk 1 above)
[the remark this answers is not in the document]
Comment (frankinspace) [reply test/net2cog/net2cog_Regression.ipynb:138]: oops that's embarrassing

**Review by flamingbear — APPROVED** [review]
Thanks for the updates, this all looks good to go. I was able to run the tests against UAT with the built docker image and the `run_notebooks.sh` script. 
But keep an eye on hanging tests? in case [this happens more than once](https://harmony.uat.earthdata.nasa.gov/admin/workflow-ui/6560011f-4327-4de4-a807-1b8a87dc4a98?fromDateTime=&toDateTime=&dateKind=createdAt&tzOffsetMinutes=&jobsLink=https%3A%2F%2Fharmony.uat.earthdata.nasa.gov%2Fadmin%2Fworkflow-ui%3Fpage%3D1%26limit%3D10)

**Review by chris-durbin — APPROVED** [review]

**Pull request merged**